### PR TITLE
doc: put Java software's build-time dependencies into nativeBuildInputs instead of into buildInputs

### PR DESCRIPTION
**Pull request by das-g**
Put Java software's build-time dependencies into `nativeBuildInputs` instead of into `buildInputs` in the examples of the NixPkgs manual.

###### Motivation for this change

`jdk`, `ant`, `gcj` and `makeWrapper` aren't needed at run-time by most Java software, even when they're being used for building the software.

See conversation at https://github.com/NixOS/nixpkgs/pull/56893#discussion_r303222522

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by FRidh — APPROVED** [review]

**Comment by das-g** [comment]
Thanks for the review, @FRidh. :smiley:

What happens now? Does this need another approval before being merged?

**Comment by FRidh** [comment]
Ideally, you, as contributor, ask the maintainers of this piece to review. Who is involved in the Java infrastructure, cc them.

**Comment by das-g** [comment]
> Ideally, you, as contributor, ask the maintainers of this piece to review. Who is involved in the Java infrastructure, cc them.

Um ... kinda-still-a-newbie here, so please help me out. Do NixPkgs manual sections even _have_ maintainers? If so, how do I find out who they are for a given section? The Python section seems to be the only one listed in [`.github/CODEOWNERS`](https://github.com/NixOS/nixpkgs/blob/master/.github/CODEOWNERS). And without intending offense to the maintainers of this section if there actually are some, but the [history](https://github.com/NixOS/nixpkgs/commits/master/doc/languages-frameworks/java.xml) of the source file doesn't seem to indicate any dedicated maintainer activity, except for whole-manual cleanups that aren't specific to that section.

So whom specifically am I supposed to contact?

**Comment by FRidh** [comment]
You are right, it seems to be unmaintained. Well, let's do something about then.

**Pull request merged**
